pkg/apis/hive/v1/agent/status.go

```diff
@@ -0,0 +1,44 @@
+package agent
+
+// InstallStrategyStatus defines the observed state of the Agent install strategy for this cluster.
+type InstallStrategyStatus struct {
+
+	// AgentNetworks provides an aggregate view of all network CIDRs detected across all Agents.
+	// It can be used to help select a valid machine network in spec if one is not known beforehand.
+	AgentNetworks []AgentNetwork `json:"agentNetworks,omitempty"`
+
+	// ControlPlaneAgentsDiscovered is the number of Agents currently linked to this ClusterDeployment.
+	// +optional
+	ControlPlaneAgentsDiscovered int `json:"controlPlaneAgentsDiscovered,omitempty"`
+	// ControlPlaneAgentsDiscovered is the number of Agents currently linked to this ClusterDeployment that are ready for use.
+	// +optional
+	ControlPlaneAgentsReady int `json:"controlPlaneAgentsReady,omitempty"`
+	// WorkerAgentsDiscovered is the number of worker Agents currently linked to this ClusterDeployment.
+	// +optional
+	WorkerAgentsDiscovered int `json:"workerAgentsDiscovered,omitempty"`
+	// WorkerAgentsDiscovered is the number of worker Agents currently linked to this ClusterDeployment that are ready for use.
+	// +optional
+	WorkerAgentsReady int `json:"workerAgentsReady,omitempty"`
+
+	ConnectivityMajorityGroups string `json:"connectivityMajorityGroups,omitempty"`
+}
+
+// AgentNetwork contains infromation about a detected host network on the agents for this cluster.
+type AgentNetwork struct {
+
+	// MachineNetwork contains the agent network CIDR.
+	MachineNetwork MachineNetworkEntry `json:"machineNetwork"`
+
+	// AgentRefs contains the namespace and name of each Agent resource associated with the cluster.
```

Comment:
Is this the set of agents which currently match the agent label selector?

And the list becomes empty when the install completes?

And it's an aggregation of data available on the Agent CRs?

And its primary use is to help you complete a required field (machine network) in this CR before the CR can be reconciled?

That all seems pretty strange, no?

Reply:
> Is this the set of agents which currently match the agent label selector?

It would be the set that match the label selector, and the machine network here.

> 
> And the list becomes empty when the install completes?

We should probably wipe it all out once machine network is set, or the cluster is marked installed.

> 
> And it's an aggregation of data available on the Agent CRs?

My understanding was just a cidr and all agents that match it.

> 
> And its primary use is to help you complete a required field (machine network) in this CR before the CR can be reconciled?
> 
> That all seems pretty strange, no?

This is definitely not a pattern I'm familiar with. Is it a use case we need to satisfy with the kube API? Or could we assume users must give us a machine network before we start, and we don't need to provide this tooling to help you find one? 

Is use of a default machine network like openshift-install does out of the question for metal clusters? I'm guessing it is.

Reply:
I agree with the sentiments, and we may be able to get by without this pattern at all. The high-level idea is that each agent discovers which subnets it is connected to, then those get reduced on the cluster resource as a list of unique subnets available on all related agents, thus making it easy for the user to pick one. It's useful in the interactive wizard UX, but not so useful for a declarative (or especially gitops) approach to cluster creation.

@avishayt @ronniel1 WDYT about omitting this list of networks for now?

Reply:
If it's expected from the user to aggregate this information from the hosts then it's ok.
You are correct that it was added as a UX feature to help the users select the right network

Reply:
Decision taken to remove this, it's too imperative and that's not our primary use case right now. We will require the machine cidr at creation time.